monteCarloTreeSearch.c

```diff
@@ -0,0 +1,383 @@
+* code that will run the game Othello and determine the best move to be made.
+* Compile using:
+* gcc -lm -Wall -o monteCarloTreeSearch monteCarloTreeSearch.c board.c node.c
+
+*/
+
+#include <time.h>
+#include <stdlib.h>
+#include <stdio.h> 
+#include <math.h>
+#include "node.h"
+#include "board.h"
+
+
+void pickMove(struct board* gameBoard, int nodeLimit, int timeLimit, int* move)
+{
+	node root; 				//root node
+	node parentNode;           //node to hold the parentNode for backPropagation
+	node newNode;              // tempNode for looping
+	node selected;				//selected value for expansion
+	node child;				//child node pointer
+
+	struct board boardState;  //gameboard pointer for use in simulation    
+
+
+	int allActions[200];           //list of all actions returned by getMoves
+	int action[2];				//action to be used for expansion
+	int boardMove[2];            // boardMove for simulation
+
+	int randomValue;			//randomValue stores generated value used to pick moves for simulation
+	double timeElapsed;			//used when calculated time taken to complete a loop
+	int gameContinues;			//used to tell if game is still being simulated 
+	int numNewNodes = 0;		//keeps track of the number of nodes created
+	int winner;					//used for determining the winner
+	int numWins = 0; 			//used to tally the wins for backpropagation
+	int numSims = 0;			//used to tally simulations for back-propagation
+	int i;						//loop value
+	int j;						//loop value
+  	int index;					//used for finding the index of the move to be made 
+  	int flag = 10;				//used if there are no moves to be made by the player
+	int numOfActions;			//used to track the number of moves available in simulation phase
+
+	double uctValue;			//used when calculating the node to expand
+	double bestValue;			//used to store the best value for expansion
+	time_t startTime, checkTime; //used to calculate the time that has passed
+
+	time(&startTime);				//track time limit
+	initializeRoot(&root, gameBoard);	  // initialize root  
+	newNode = root;  					// temporary newNode
+	selected = root;					//initialize selected value for expansion
+
+	while(numNewNodes < nodeLimit && flag > 0)
+	{
+		gameContinues = 100;	//initialize the simulation phase gameloop value
+		bestValue = -1000;		// initialize bestvalue as a negative number
+		numOfActions = 0;		//set numOfActions to zero each loop through
+
+		//1. Select a node using UCT and the functions provided by #1. (See the js implementation of 
+		//selection and the js UCT implementation as a guide).
+
+		printf("Selection\n");
+		//UCT calculation to select next node 
+		for(i = 0; i < newNode.numChildren; i++)
+		{
+			child = *newNode.children[i];
+			//UCT calculation
+			uctValue = child.numWins / child.numSimulations + 
+			sqrt(2 * log(newNode.numSimulations / child.numSimulations));
+
+			//check if the uct value for each child is better than the one before
+			if(uctValue > bestValue)
+			{
+				selected = child;
```

Comment:
Make sure "selected" is a pointer so we're operating on the actual tree nodes and not copies of the nodes.